core-rust/state-manager/src/state_manager.rs

```diff
@@ -477,25 +432,44 @@ where
         };
 
         //========================================================================================
-        // PART 4:
+        // PART 3:
         // We continue by attempting to execute the remaining transactions in the proposal
         //========================================================================================
 
         let mut rejected_transactions = Vec::new();
         let pending_transaction_timestamp = SystemTime::now();
         let mut pending_transaction_results = Vec::new();
 
-        for (i, proposed_prepare_result) in prepared_proposed_results.into_iter().enumerate() {
+        for (index, raw_user_transaction) in prepare_request
+            .proposed_transactions
+            .into_iter()
+            .enumerate()
+        {
             // Don't process any additional transactions if next epoch has occurred
             if next_epoch.is_some() {
                 break;
             }
 
-            let (raw, prepared) = match proposed_prepare_result {
-                Ok(prepared) => prepared,
+            let prepare_results = LedgerTransaction::from_raw_user(&raw_user_transaction)
+                .map_err(|err| {
+                    TransactionValidationError::PrepareError(PrepareError::DecodeError(err))
+                })
+                .and_then(|ledger_transaction| {
+                    ledger_transaction.to_raw().map_err(|err| {
+                        TransactionValidationError::PrepareError(PrepareError::EncodeError(err))
+                    })
+                })
+                .and_then(|raw_ledger_transaction| {
+                    self.ledger_transaction_validator
+                        .prepare_from_raw(&raw_ledger_transaction)
```

Comment:
You can combine these two `and_then`s by using `.prepare_from_model`

Reply:
I cannot :(
The `raw_ledger_transaction` (i.e. the mid-step result) needs to be returned, since it is later used (as `CommittableTransaction.raw`).